src/vehicle_use.cpp

```diff
@@ -1744,6 +1744,129 @@ void vehicle::use_harness( int part, const tripoint &pos )
     }
 }
 
+std::vector<item_count_tuple> vehicle::get_comestible_liquids()
+{
+    std::vector<item_count_tuple> liquids;
+    std::map<item *, uint32_t> liquids_map;
+
+    liquids.clear();
+    liquids_map.clear();
+
+    for( auto &part : parts ) {
+        const int remaining = part.ammo_remaining();
+
+        if( !part.is_tank() || part.base.contents_made_of( SOLID ) || remaining <= 0 ) {
+            continue;
+        }
+
+        item *food_ptr = part.base.get_food();
+
+        if( food_ptr == nullptr || !food_ptr->is_comestible() ) {
+            continue;
+        }
+
+        const islot_comestible &comestible = *food_ptr->get_comestible();
+
+        if( "DRINK" != comestible.comesttype ) {
+            continue;
+        }
+
+        const itype_id liquid_id = part.ammo_current();
+
+        if( liquid_id == "null" || liquid_id == "water" ) {
+            continue;
+        }
+
+        liquids_map[food_ptr] += static_cast<uint32_t>( remaining );
+    }
+
+    for( auto &liquid : liquids_map ) {
+        liquids.emplace_back( liquid.first, liquid.second );
+    }
+
+    std::sort( liquids.begin(), liquids.end(),
+    []( const item_count_tuple & a, const item_count_tuple & b ) {
+        return std::get<1>( a ) > std::get<1>( b );
+    } );
+
+    return liquids;
+}
+
+item_count_tuple vehicle::select_comestible_liquid( const std::vector<item_count_tuple> &liquids,
+        const std::string &message )
+{
+    uilist selectmenu;
+    int choice = std::numeric_limits<int>::min();
+    int idx = 0;
+
+    if( liquids.empty() ) {
+        for( auto &liquid : liquids ) {
+            const item *p_item = std::get<0>( liquid );
+            selectmenu.addentry( idx++, true, MENU_AUTOASSIGN, p_item->tname( std::get<1>( liquid ) ) );
+        }
+
+        if( selectmenu.entries.size() == 1 ) {
+            choice = selectmenu.entries.front().retval;
+        } else {
+            selectmenu.text = message;
+            selectmenu.query();
+            choice = selectmenu.ret;
+        }
+
+        if( choice >= 0 && choice < idx ) {
+            try {
+                return liquids[static_cast<size_t>( choice )];
+            } catch( ... ) {
```

Comment:
It's better to avoid catch-alls and specify the exact exception you're trying to catch.

Reply:
Fixed.